### PR TITLE
[mlir] Use arith max or min ops instead of cmp + select

### PR DESCRIPTION
**Pull request by mlevesquedion**
I believe the semantics should be the same, but this saves 1 op and simplifies the code.

For example, the following two instructions:

```
%2 = cmp sgt %0, %1
%3 = select %2, %0, %1
```

Are equivalent to:

```
%2 = maxsi %0 %1
```

### CODE REVIEW

**Comment by makslevental** [comment]
@RoboTux @banach-space @matthias-springer @Mogball @ThomasRaoux I tagged you because you're (roughly) the most recent editors (according to `git blame`) feel to remove/ignore if you're not up for reviewing.

**Review by dcaballe — COMMENTED** [review]
LGTM. I'm having a dejavu, though... as if somebody else had already tried something similar in the Affine to Standard lowering and we realized that there was some semantic differences/corner cases... Could you please do a search on Phabricator to be sure that there isn't anything similar?

File: mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp [hunk 1]
```diff
@@ -48,13 +43,17 @@ static Value buildMinMaxReductionSeq(Location loc,
...
+  assert(predicate == arith::CmpIPredicate::sgt ||
+         predicate == arith::CmpIPredicate::slt);
 
   auto valueIt = values.begin();
   Value value = *valueIt++;
   for (; valueIt != values.end(); ++valueIt) {
-    auto cmpOp = builder.create<arith::CmpIOp>(loc, predicate, value, *valueIt);
-    value = builder.create<arith::SelectOp>(loc, cmpOp.getResult(), value,
-                                            *valueIt);
+    if (predicate == arith::CmpIPredicate::sgt) {
```
Comment (Mogball) [inline mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp:52]: style: drop trivial braces

**Review by Mogball — APPROVED** [review]

**Comment by mlevesquedion** [comment]
> LGTM. I'm having a dejavu, though... as if somebody else had already tried something similar in the Affine to Standard lowering and we realized that there was some semantic differences/corner cases... Could you please do a search on Phabricator to be sure that there isn't anything similar?

Thanks for taking the time to review! I'm actually not sure how to search on Phabricator. When I search "llvm phabricator" on Google I land on this page: https://reviews.llvm.org/, which does not seem to provide a way to search old changes. Do you know how to search on Phabricator? I asked on Discourse: https://discourse.llvm.org/t/how-to-search-old-pull-requests-on-phabricator/77133. (I also tried a search on GitHub and got no hits.)

Alternatively, are there additional checks/tests that I could add/do to validate this change?

**Review by dcaballe — APPROVED** [review]
Thanks for checking. I also gave it a try and couldn't find anything related to affine min/max so I could be related to other ops... Let's go with this.

File: mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp (hunk 1 above)
Comment (mlevesquedion) [reply mlir/lib/Conversion/AffineToStandard/AffineToStandard.cpp]: Thanks for the review! Done.

**Comment by mlevesquedion** [comment]
Thanks for the reviews, folks! I've addressed the review feedback, can you merge this for me? (I don't have commit access.)

**Pull request merged**